File: src/com/company/Reverser.java
package com.company;

public class Reverser {

    private String input;
    private String output;

    public Reverser(String input) {
        this.input = input;
        output = "";
    }

    public String getInput() {
        return input;
    }

    public String getOutput() {
        //this.reverse();
        return output;
    }

    public void reverse()
    {
        int size = input.length();
        StackX stackX = new StackX(size);
        for (int i = 0; i < size; i++)
        {
            char inputChar = input.charAt(i);
            stackX.peek(inputChar);
        }
        while (!stackX.isEmpty())
        {
            char outputChar = stackX.pop();
            output += outputChar;
        }
    }
}

File: StructuresOnKotlin/src/Reverser.kt
class Reverser(private val value:String) {

    private var output = ""

    fun doRev():String
    {
        val stackArray: StackXChar = StackXChar(value.length)
        var ch:Char
        for(element in value)
            stackArray.push(element)
        while (!stackArray.isEmpty())
        {
            ch = stackArray.pop()
            output += ch
        }
        return output
    }
}